manifests/service.pp

```diff
@@ -3,21 +3,68 @@
   validate_string($service_name)
   validate_bool($service_enable)
 
+  $surplus_service_worker = $service_workers + 1
+
   if !($service_ensure in ['running', 'stopped']) {
     fail('service_ensure parameter must be running or stopped')
   }
 
   if $service_manage {
-    servicetools::install_systemd_unit { $service_name:
-      service_options => {
-        'Type'      => 'forking',
-        'ExecStart' => "/usr/bin/php ${site_root}/framework/cli-script.php dev/resque/run queue=* count=${service_workers} flush=1",
-        'User'      => 'www-data',
-        'Restart'   => 'on-failure'
-      },
-      service_ensure  => $service_ensure,
-      service_enable  => $service_enable
+
+    # @ is required to be able to launch multiple aliased copies of the same service.
+    file { "/etc/systemd/system/$service_name@.service":
+      ensure => file,
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644',
+      content => template("deploynaut/systemd_service.erb"),
+      notify => Exec['deploynaut systemctl daemon reload'],
+    } -> file { "/etc/systemd/system/$service_name.target":
+      ensure => file,
+      owner  => 'root',
+      group  => 'root',
+      mode   => '0644',
+      content => template("deploynaut/systemd_target.erb"),
+      notify => Exec['deploynaut systemctl daemon reload'],
+    }
+
+
+    # Manage N workers through systemd, so that we can get systemd to gracefully restart them.
+    if ($service_enable or $service_ensure=='running') {
+      # We also enable these units if "running" is required because only $service_name.target depends/wants them,
+      # so they won't run on boot anyway.
+      exec { "disable surplus deploynaut workers $service_workers":
```

Comment:
Try to flip the dependency - instead of PartOf and WantedBy, add the dependency to the .target. And maybe use StopWhenUnneeded=true, so that the no-longer-needed workers don't loiter?